### PR TITLE
fix: mempool endpoints errors should provide better UX

### LINKED ISSUE #89: Application and mempool endpoints errors should provide better UX
We have multiple application and mempool related endpoints. Errors in most of them will make the application unusable.
For example, if an unavailability or error of an endpoint prevents staking, we should disable the staking form.

Some mempool findings:
```
# btcTipHeightUrl is down

Staking has not yet started https://i.imgur.com/icKTBFs.png

# addressInfoUrl is down

Used in getAddressBalance, nothing will happen since we calculate the address balance using confirmed UTXO https://i.imgur.com/uzDzUUs.png

# utxosInfoUrl is down

Balance is constantly loading, cannot stake: https://i.imgur.com/MfM87Di.png

# networkFeesUrl is down

Cannot stake due that we cannot load the fees https://i.imgur.com/qyx7tk3.png

# validateAddressUrl is down

Balance is constantly loading (because used in `getFundingUTXOs`), cannot stake: https://i.imgur.com/k2mvfhR.png

# txInfoUrl is down

If `maxDelegationPendingDuration` matched, then the delegation is excluded from the local storage `validDelegations` array, assuming that the transaction is no longer pending or valid.
Leads to clearing out the local storage pending delegation

# pushTxUrl is down

Will not be able to broadcast the transaction
```

### PR DESCRIPTION
**Pull request by totraev**
Closes #89

### CODE REVIEW

**Comment by totraev** [comment]
## btcTipHeightUrl is down

###Before
Screenshot 2024-09-19 at 17 30 31

###After
...

**Comment by totraev** [comment]
## utxosInfoUrl is down

### Before
Screenshot 2024-09-19 at 17 35 08

### After

**Comment by totraev** [comment]
## networkFeesUrl is down

### Before
Screenshot 2024-09-19 at 17 55 42

### After
...

**Comment by totraev** [comment]
## validateAddressUrl is down

### Before
Screenshot 2024-09-19 at 17 35 08

### After
...

**Comment by totraev** [comment]
## addressInfoUrl is down

Used in getAddressBalance, nothing will happen since we calculate the address balance using confirmed UTXO

**Comment by totraev** [comment]
## pushTxUrl is down

Will not be able to broadcast the transaction, but error modal will be displayed. Nothing feels really broken

**Comment by totraev** [comment]
## txInfoUrl is down

If `maxDelegationPendingDuration` matched, then the delegation is excluded from the local storage `validDelegations` array, assuming that the transaction is no longer pending or valid.
Leads to clearing out the local storage pending delegation after 24 hours

File: src/app/components/Staking/Staking.tsx [hunk 1]
```diff
@@ -136,7 +139,7 @@ export const Staking: React.FC<StakingProps> = ({
 
   // load global params and calculate the current staking params
   const globalParams = useGlobalParams();
-  useMemo(() => {
+  useEffect(() => {
```
Comment (gbarkhatov) [inline src/app/components/Staking/Staking.tsx:142]: Are we okay using `useEffect` without the dependencies array? Any future concerns with the multiple re-rendering or performance?

**Review by gbarkhatov — APPROVED** [review]
LGTM

File: src/app/components/Staking/Staking.tsx (hunk 1 above)
Comment (totraev) [reply src/app/components/Staking/Staking.tsx:142]: @gbarkhatov sorry, I don't follow, this particular `useEffect` has dependency array or do I miss something?

**Review by gbarkhatov — APPROVED** [review]

File: src/app/components/Staking/Staking.tsx (hunk 1 above)
Comment (gbarkhatov) [reply src/app/components/Staking/Staking.tsx:142]: Nvm

**Review by jeremy-babylonlabs — APPROVED** [review]

**Pull request merged**